tests/python/pants_test/engine/test_rules.py

```diff
@@ -106,6 +107,35 @@ def test_creation_fails_with_bad_declaration_type(self):
       RuleIndex.create([A()])
 
 
+class RuleArgumentAnnotationTest(unittest.TestCase):
+  def test_name_kwarg(self):
+    @rule(name='A named rule')
+    def named_rule(a: int, b: str) -> bool:
+      return False
+    self.assertIsNotNone(named_rule.rule)
+    self.assertEqual(named_rule.rule.name, "A named rule")
+
+  def test_bogus_rule(self):
+    with self.assertRaises(UnrecognizedRuleArgument):
+      @rule(bogus_kwarg='TOTALLY BOGUS!!!!!!')
+      def named_rule(a: int, b: str) -> bool:
+        return False
+
+  def test_console_rule_automatically_gets_name_from_goal(self):
+    @console_rule
+    def some_console_rule(console: Console) -> Example:
+      yield Example(exit_code=0)
```

Comment:
As of a few hours ago, this can and I think will soon need to be `return` thanks to our new `async await` style.

Reply:
will this break with the current state of @cosmicexplorer 's changes pertaining to async/await? if not I'd rather get this patch set merged without changing it and make the change to `return` as part of whatever large commit we do to do that across the whole codebase.

Reply:
It's now safe to use async/await, along with the old yield style. But once https://github.com/pantsbuild/pants/pull/8652 gets merged, this test will fail. So, I recommend changing this to `return` now because it will still succeed and then you reduce the risk of merging before Danny and causing his PR to then have issues.